clang/lib/AST/CanonBounds.cpp

```diff
@@ -180,17 +180,26 @@ Lexicographic::CompareScope(const DeclContext *DC1, const DeclContext *DC2) cons
 }
 
 Result
-Lexicographic::CompareDecl(const CapturedDecl *CD1,
-                           const CapturedDecl *CD2) const {
+Lexicographic::CompareDecl(const CapturedDecl *CD1Arg,
+                           const CapturedDecl *CD2Arg) const {
+  const CapturedDecl *CD1 = dyn_cast<CapturedDecl>(CD1Arg->getCanonicalDecl());
+  const CapturedDecl *CD2 = dyn_cast<CapturedDecl>(CD2Arg->getCanonicalDecl());
+  if (CD1 == CD2)
```

Comment:
Do we want to return `Result::Equal` even if both `CD1` and `CD2` are null? If not, then the null checks from line 190 should move here.

Reply:
This is a fair question. For the sake of uniformity, I have written the code to execute in the same way that the other overloaded function `CompareDecl` on line 206 (the function that compares two `NamedDecls`) executes.